enhancements/observability/cluster-observability-operator-logging-stack.md

```diff
@@ -0,0 +1,405 @@
+2. COO creates a `ClusterLogForwarder` resource with at least an output of type `lokistack` (See [API Extentions](#api-extensions)).
+3. COO creates a deployment for [logging-view-plugin](logging-view-plugin) when reconciling the `UIPlugin` resource owned by `LoggingStack`.
+4. COO patches the `Consoles` resource when reconciling the `UIPlugin` resource owned by `LoggingStack`.
+
+### Cluster Obserbability Operator design impact managing OLM-based operators
+
+When an operator (i.e. COO) managing the installation/upgrade of further operators (i.e. CLO and LO) as well as resources owned by the managed operators, they key consideration should be on when managed CRDs will registered on the Kube APIserver. The registration event is a cornerstone for the kubernetes clientsets used in COO to reconcile shared custom resources such as `ClusterLogForwarder` and `Lokistack`. In particular it is mandatory for client abstration such as `informers` and `listers` to execute **only** on resource known by the Kube APIserver.
+
+Therefore the COO requires a major design overhaul that supports the following aspects:
+1. Independent reconciliation loops for installing/upgrading managed operators.
+2. Reconciliation loops for shared resources such as `ClusterLogForwarder` and `Lokistack` only and only if registration of the corresponding CRD successful
+3. Robust signaling between the above two and in turn mindful setup and tear-down of underlying kubernetes clientsets/informers/listers.
+
+In summary the COO binary will hold two actively communicating sub-controllers each managing distinct aspects for an owned resource like `LoggingStack`, i.e.:
+1. **Operators-Controller**: This controller reconciles a `OperatorConfig` resource into `OperatorGroup` and `Subscription` resources for CLO and LO only. It tracks success as well as failure conditions for each operator installation/upgrade phase on the `LoggingStack` status field.
+2. **Components-Controller**: This controller reconciles the same `LoggingStack` resource into the resources `ClusterLogForwarder`, `LoggingStack` and `UIPlugin` that in turn install and manage the logging stack components (e.g. collection, storage and visualization plugin).
+
+### LoggingStack status conditions
+
+To effectively track the success/failure status at any point in time of the managed operators (i.e. CLO and LO), the following conditions will be aggregated from OLM resources into the `OperatorConfig` status conditions:
+1. TBD
+2. ...
+
+To effectively track the success/failure status of the managed but shared custom resources, the following conditions will be aggregated from the `ClusterLogForwarder` and `LokiStack` resources into the `LoggingStack` status conditions:
+1. TBD
+2. ...
+
+Finally the following conditions will be managed in the `LoggingStack` status conditions about visulization and correlation:
+1. TBD
+2. ...
+
+### Automated korrel8r deployment management
+
```

Comment:
NOTE: [Multiple stores per domain, robust to store failures. \* Issue #112 \* korrel8r/korrel8r](https://github.com/korrel8r/korrel8r/issues/112)
This simplifies configuration of korrel8r: the COO should provide a configuration including all stores that _might_ be available, korrel8r will use whatever it can connect to successfully, and will not fail due to unavailable stores. In the simplest case we can use a static configuration file including all the "standard" places where obs. stores are usually installed. The COO can provide a more accurate configuration based on the components it manages.